### PR TITLE
Add JSON Lib

### PR DESCRIPTION
**Pull request by thojkooi**
Adds an JSON library implementation.

TODO:
- [ ] Character checking values should be number constants
- [x] Validate edge cases
- [ ] Write unit tests?
- [ ] Serialize RV data types
- [ ] Expand code with more comments
#### JSON.parse

Parse a JSON string to a Carma object.

**Parameters:**
- 0: _string_

**Returns:**
- _Carma object_
#### JSON.stringify

Convert a Carma object to a JSON string

**Parameters:**
- _0: Carma Object_

**Returns:**
- _string_

---
#### Carma objects to JSON string

Convert any Carma object to a [JSON](http://www.json.org/) string using `JSON.stringify`.

``` javascript
myObject = new JSONObject();
myObject.name = "object";

myObject.data = [new JSONObject(), new JSONObject(), new JSONObject()];
{ 
    _x{"id"} = _forEachIndex; 
    _x{"value"} = (random 100); 
} forEach myObject.data;
private _jsonString = JSON.stringify(myObject);
```

The above code outputs a string with the following contents:

``` javascript
{"name":"object","data":[{"id":0,"value":14.0339},{"id":1,"value":44.513},{"id":2,"value":78.4415}]}
```
#### Parse a JSON String to Carma object

JSON.parse returns a hashmap object with all keys set to the values from the JSON string.
All functionality normally available to a hashmap is also available to JSON objects obtained through `JSON.parse`.

``` javascript
private _objRet = JSON.parse('{"key": 1, "myArray": ["valueOne", "value2", 4], "testObj": {isValid: true, "date": "12/02/2015"}}');

// Check if variable exists
if (_objRet._hasKey("key")) then {
    // Do stuff
};

// Use variables from object
if (_objRet.testObj.isValid) then {
    // Do things
};
```

### CODE REVIEW

**Comment by Glowbal** [comment]
### JSON parsing

#### Carma objects to JSON string
Convert any Carma object to a [JSON](http://www.json.org/) string using `JSON.stringify`.
```javascript
myObject = new JSONObject();
myObject.name = "object";

myObject.data = [new JSONObject(), new JSONObject(), new JSONObject()];
{ 
    _x{"id"} = _forEachIndex; 
    _x{"value"} = (random 100); 
} forEach myObject.data;
private _jsonString = JSON.stringify(myObject);
```
The above code outputs a string with the following contents:
```javascript 
{"name":"object","data":[{"id":0,"value":14.0339},{"id":1,"value":44.513},{"id":2,"value":78.4415}]}
```

#### Parse a JSON String to Carma object
JSON.parse returns a hashmap object with all keys set to the values from the JSON string.
All functionality normally available to a hashmap is also available to JSON objects obtained through `JSON.parse`.
```javascript
private _objRet = JSON.parse('{"key": 1, "myArray": ["valueOne", "value2", 4], "testObj": {isValid: true, "date": "12/02/2015"}}');

// Check if variable exists
if (_objRet._hasKey("key")) then {
    // Do stuff
};

// Use variables from object
if (_objRet.testObj.isValid) then {
    // Do things
};
```

File: rv/addons/lib/lib_carma2_json.sqf [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+            } forEach _value;
+            _ret + "]";
+        };
+        case ("CODE"): {''}; // exclude code and functions from json string
+        case ("OBJECT"): {
+            private _object = new JSONObject();
+            private _objInfo = new JSONObject();
+
+            {
+                _object setvariable [_x, _value getvariable _x];
```
Comment (NouberNou) [inline rv/addons/lib/lib_carma2_json.sqf]: You can get/set dynamic members on objects using the `{}` operator. For example this line would be: `_object{_x} =  _value{_x};` This also works on any object that can get/setVariable.

**Comment by thojkooi** [comment]
Fixed the `get/setVariable`.
